### PR TITLE
FHAC-946: fix codes instead of mitigate through document

### PR DESCRIPTION
**Pull request by mhpnguyen**
Fix codes instead of mitigate through document.
@TabassumJafri @danielrf123: Can you please review this?

### CODE REVIEW

**Comment by mhpnguyen** [comment]
SonarQube analysis reported 1 issue:
- MAJOR 1 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
- MAJOR [Use isEmpty() to check whether the collection is empty or not.](https://github.com/CONNECT-Solution/CONNECT/blob/40866aa395e71c5cab8be38a4502f4bec3f5cb74/Product/Production/Services/PatientDiscoveryCore/src/main/java/gov/hhs/fha/nhinc/patientdiscovery/model/builder/impl/PatientSearchResultsModelBuilderImpl.java#L153) [rule](http://GFE-ONC-LDEV15.connect.org:9000/coding_rules#rule_key=squid%3AS1155)

File: Product/Production/Services/PatientDiscoveryCore/src/main/java/gov/hhs/fha/nhinc/patientdiscovery/model/builder/impl/PatientSearchResultsModelBuilderImpl.java [hunk 1]
```diff
@@ -110,28 +114,26 @@ private void extractAndAddPatient(PRPAMT201310UV02Patient msgPatient) {
...
-                        if (!firstNameFound) {
-                            patient.setFirstName(((EnExplicitGiven) nameValue).getContent());
-                            firstNameFound = true;
-                        } else if (!secondNameFound) {
-                            patient.setMiddleName(((EnExplicitGiven) nameValue).getContent());
-                            secondNameFound = true;
+                        //the first given name is first name, following by middle name
+                        String givenName = ((EnExplicitGiven) nameValue).getContent();
+                        if (StringUtils.hasText(patient.getFirstName())){
+                            patient.setMiddleName(givenName);
```
Comment (danielrf123) [inline Product/Production/Services/PatientDiscoveryCore/src/main/java/gov/hhs/fha/nhinc/patientdiscovery/model/builder/impl/PatientSearchResultsModelBuilderImpl.java:127]: Did we decide that we need a middle name for the Patient?

Comment (mhpnguyen) [inline Product/Production/Services/PatientDiscoveryCore/src/main/java/gov/hhs/fha/nhinc/patientdiscovery/model/builder/impl/PatientSearchResultsModelBuilderImpl.java:127]: We do need middle name.  However, if the given name list has two entry, the first one is first name and second one is middle name

**Comment by danielrf123** [comment]
I endorse this PR for a merge.

**Comment by TabassumJafri** [comment]
Reviewed PR and good with merge.

**Comment by TabassumJafri** [comment]
merged

**Pull request merged**